certbot/certbot/_internal/configuration.py

```diff
@@ -138,6 +142,27 @@ def renewal_post_hooks_dir(self):
         return os.path.join(self.renewal_hooks_dir,
                             constants.RENEWAL_POST_HOOKS_DIR)
 
+    @property
+    def key_type(self):
```

Comment:
I think you do not need this function to compute the `key_type` dynamically in fact. I believe you added it to answer to the comment I wrote in https://github.com/certbot/certbot/pull/8254#discussion_r510807545.

The situation to cover here is a bad behavior between `--force-renewal` and `--reuse-key` when `certbot run` or `certbot certonly` are run. In these precise cases, the renewal configuration parameters related to the key is lost because of the way it is implemented in the `renewal` module.

The following code has been written (and updated by you) to cover this corner case.
https://github.com/atombrella/certbot/blob/aa1950347d225e3067771a40cc397f4af0bcb765/certbot/certbot/_internal/renewal.py#L504-L513

To completely fix the issue (by storing `key_type`, you can simply modify the function `_update_renewal_params_from_key`. What you should to is to set `config.key_type = 'rsa'` in case of an RSA key, and `config.key_type = 'ecdsa'` in case of an ECDSA key.

Reply:
OK. Done.